src/Config.php

```diff
@@ -73,10 +80,21 @@ class Config
      */
     public function __construct(string $path, bool $fileExists = false, array $settings = [])
     {
+        $pluginSettings = $settings['plugins'] ?? [];
+        unset($settings['plugins']);
+
         $this->path       = $path;
         $this->fileExists = $fileExists;
         $this->settings   = $settings;
 
+        foreach ($pluginSettings as $plugin) {
+            $name = $plugin['plugin'];
+            $options = isset($plugin['options']) && is_array($plugin['options'])
+                ? $plugin['options']
+                : [];
+            $this->plugins[$name] = new Config\Plugin($name, $options);
```

Comment:
Why do you create here new object it couple the call to Plugin. Factory pattern an interface is maybe better to reduce coupling here.  From point of view, a function `getPlugins` would better constructor should not have business logic.

Reply:
I followed the same pattern that `Config\Hook` used just below these lines: https://github.com/captainhookphp/captainhook/blob/154aea58b13e1c5e5005aad04270a94ccb010fe9/src/Config.php#L80-L82